leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/CaliforniumEndpointsManager.java

```diff
@@ -271,9 +271,14 @@ public synchronized ServerIdentity createEndpoint(ServerInfo serverInfo, boolean
                         "Unable to create endpoint for %s using OSCORE : Unable to decode OSCORE AEAD from %s.",
                         serverInfo, serverInfo.oscoreSetting), e);
             }
+
+            // TODO OSCORE kind of hack because californium doesn't support an empty byte[] array for salt ?
+            byte[] masterSalt = serverInfo.oscoreSetting.getMasterSalt().length == 0 ? null
+                    : serverInfo.oscoreSetting.getMasterSalt();
```

Comment:
I currently testing to update my previous OSCORE demo PRs with our last modification.

And I just saw that current californium API does not like if masterSalt is an empty byte array.
I guess this comes from that codes from : [org.eclipse.californium.oscore.OSCoreCtx.OSCoreCtx` line 261](https://github.com/eclipse/californium/blob/75e72b207eb927e914ac0062faac8009b7e4c55c/cf-oscore/src/main/java/org/eclipse/californium/oscore/OSCoreCtx.java#L261-L266)
```java
if (master_salt == null) {
	// Default value. Automatically initialized with 0-es.
	this.common_master_salt = new byte[this.kdf.getKeySize() / Byte.SIZE];
} else {
	this.common_master_salt = master_salt.clone();
}
```
But reading the OSCORE [RFC8613](https://datatracker.ietf.org/doc/html/rfc8613#section-3.2) : 
>  Master Salt
>      *  Default is the empty byte string

Is it a kind of bug or is it the expected behavoir ?

Reply:
Looking into things a bit this is my understanding. The default Master Salt in OSCORE should indeed be the empty byte string. However, later when that is fed into the HKDF, if the salt is zero length, it should be padded:
https://datatracker.ietf.org/doc/html/rfc5869#section-2.2
```
Inputs:
      salt     optional salt value (a non-secret random value);
               if not provided, it is set to a string of HashLen zeros.
```

So that is why if the master salt is null that padding occurs. But yeah what seems to be missing is considering if the developer sets the empty byte string as salt in the context constructor. So it would be better if it was:
```
if (master_salt == null || master_salt.length == 0) {
	// Default value. Automatically initialized with 0-es.
	this.common_master_salt = new byte[this.kdf.getKeySize() / Byte.SIZE];
} else {
	this.common_master_salt = master_salt.clone();
}
```
(Or waiting to do the padding in the deriveKey method may be even clearer). I will add this to the list of updates to do.